src/clib/pio_msg.c

```diff
@@ -2521,7 +2521,7 @@ int pio_msg_handler2(int io_rank, int component_count, iosystem_desc_t **iosys,
         {
             my_iosys = iosys[cmp];
             LOG((1, "about to call MPI_Irecv union_comm = %d", my_iosys->union_comm));
-            if ((mpierr = MPI_Irecv(&msg, 1, MPI_INT, my_iosys->comproot, MPI_ANY_TAG,
+            if ((mpierr = MPI_Irecv(messages+cmp, 1, MPI_INT, my_iosys->comproot, MPI_ANY_TAG,
```

Comment:
How about the less confusing notation messages[cmp]?

Reply:
changed